### PR TITLE
support migrate subcommand

### PR DESCRIPTION
**Pull request by zhoney**
implemented: #35

Change-Id: Ib5493f4dc7d6b606f8e64e390df43103d51aa458

### CODE REVIEW

File: src/main/java/com/baidu/hugegraph/cmd/HugeGraphCommand.java [hunk 1]
```diff
@@ -271,6 +320,31 @@ private void checkMainParams() {
         }
     }
 
+    private static SubCommands.Backup migrate2Backup(
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/cmd/HugeGraphCommand.java]: convMigrate2Backup

File: src/main/java/com/baidu/hugegraph/cmd/HugeGraphCommand.java [hunk 2]
```diff
@@ -271,6 +320,31 @@ private void checkMainParams() {
...
+        backup.splitSize(migrate.splitSize());
+        backup.directory(migrate.directory());
+        backup.logDir(migrate.logDir());
+        backup.types(migrate.types());
+        backup.retry(migrate.retry());
+        backup.hdfsConf(migrate.hdfsConf());
+        return backup;
+    }
+
+    private static SubCommands.Restore migrate2Restore(
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/cmd/HugeGraphCommand.java]: ditto

File: src/main/java/com/baidu/hugegraph/base/HdfsDirectory.java [hunk 3]
```diff
@@ -108,6 +108,20 @@ public void ensureDirectoryExist(boolean create) {
...
     }
 
+    @Override
+    public void removeDirectory() {
+        FileSystem fs = this.fileSystem();
+        Path path = new Path(this.directory());
+        try {
+            E.checkState(fs.exists(path) &&
+                         fs.getFileStatus(path).isDirectory(),
+                         "Directory '%s' not exists", this.directory());
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/base/HdfsDirectory.java]: "The directory does not exist: '%s'"

File: src/main/java/com/baidu/hugegraph/base/LocalDirectory.java [hunk 4]
```diff
@@ -154,4 +161,16 @@ private static void ensureDirectoryExist(String directory, boolean create) {
             }
         }
     }
+
+    private static void removeDirectory(String directory) {
+        File dir = new File(directory);
+        E.checkState(dir.exists() && dir.isDirectory(),
+                     "Directory '%s' not exists", dir.getAbsolutePath());
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/base/LocalDirectory.java]: ditto

**Review by javeme — CHANGES_REQUESTED** [review]

**Review by Linary — DISMISSED** [review]

File: src/main/java/com/baidu/hugegraph/cmd/SubCommands.java [hunk 5]
```diff
@@ -122,16 +123,32 @@ private void initSubCommands() {
...
+        }
     }
 
     @Parameters(commandDescription = "Restore graph schema/data. If directory" +
                                      " is on HDFS, use -D to set HDFS params " +
                                      "if needed. For exmaple:" +
                                      "-Dfs.default.name=hdfs://localhost:9000")
-    public class Restore extends BackupRestore {}
+    public static class Restore extends BackupRestore {
+        @Parameter(names = {"--remove"},
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/cmd/SubCommands.java]: prefer --clean

File: src/main/java/com/baidu/hugegraph/cmd/SubCommands.java [hunk 6]
```diff
@@ -122,16 +123,32 @@ private void initSubCommands() {
...
+                   description = "Whether to remove graph schema/data")
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/cmd/SubCommands.java]: Whether to remove the directory of graph data after restored

File: src/main/java/com/baidu/hugegraph/cmd/SubCommands.java [hunk 7]
```diff
@@ -142,15 +159,70 @@ public String formatter() {
...
+        public String targetGraph = "hugegraph";
+
+        @Parameter(names = {"--graph-mode", "-m"}, arity = 1,
+                   converter = GraphModeConverter.class,
+                   description = "Mode used when migrating to target graph, " +
+                                 "include: [RESTORING, MERGING]")
+        public GraphMode mode = GraphMode.RESTORING;
+
+        @Parameter(names = {"--remove"},
+                   description = "Whether to remove graph schema/data")
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/cmd/SubCommands.java]: ditto

**Review by javeme — CHANGES_REQUESTED** [review]

File: src/main/java/com/baidu/hugegraph/manager/RestoreManager.java [hunk 8]
```diff
@@ -41,6 +41,7 @@
 public class RestoreManager extends BackupRestoreBaseManager {
 
     private GraphMode mode = null;
+    private boolean remove;
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/manager/RestoreManager.java]: clean

**Review by javeme — APPROVED** [review]

**Review by Linary — APPROVED** [review]

**Pull request merged**